### PR TITLE
`multiple_shoot` for `EnsembleProblem`

### PR DESCRIPTION
**Pull request by vboussange**
Hi,
This PR overloads the `multiple_shoot` function, so that it can be called with problems of type `EnsembleProblem`.
This is quite useful 😀 
It is a suggestion for issue #601.

### CODE REVIEW

File: src/multiple_shooting.jl [hunk 1]
```diff
@@ -11,27 +11,47 @@ multiple_shoot(p, ode_data, tsteps, prob, loss_function, solver, group_size;
...
+    - `tsteps`: Timesteps on which ode_data was calculated.
+    - `prob`: ODE problem that the Neural Network attempts to solve.
+    - `loss_function`: Any arbitrary function to calculate loss.
+    - `continuity_loss`: Function that takes states ``\\hat{u}_{end}`` of group ``k`` and
+    ``u_{0}`` of group ``k+1`` as input and calculates prediction continuity loss
+    between them.
+    If no custom `continuity_loss` is specified, `sum(abs, û_end - u_0)` is used.
+    - `solver`: ODE Solver algorithm.
+    - `group_size`: The group size achieved after splitting the ode_data into equal sizes.
+    - `continuity_term`: Weight term to ensure continuity of predictions throughout
```
Comment (ChrisRackauckas) [inline src/multiple_shooting.jl:34]: put this in a separate docstring on the other dispatch.

File: src/multiple_shooting.jl [hunk 2]
```diff
@@ -126,6 +146,118 @@ function multiple_shoot(
...
 end
 
+function multiple_shoot(
+    p::AbstractArray,
+    ode_data::AbstractArray,
+    tsteps::AbstractArray,
+    ensembleprob::EnsembleProblem,
+    ensemblealg::SciMLBase.BasicEnsembleAlgorithm,
+    loss_function::Function,
+    continuity_loss::Function,
```
Comment (ChrisRackauckas) [inline src/multiple_shooting.jl:156]: This stops specialization BTW

File: src/multiple_shooting.jl [hunk 3]
```diff
@@ -126,6 +146,118 @@ function multiple_shoot(
...
+        loss_function,
+        continuity_loss,
+        solver,
+        group_size;
+        continuity_term,
+        kwargs...
+    )
+    
+end
+
```
Comment (ChrisRackauckas) [inline src/multiple_shooting.jl:260]: we should really move the optional argument to last

File: src/multiple_shooting.jl [hunk 4]
```diff
@@ -126,6 +146,118 @@ function multiple_shoot(
...
+    solver::DiffEqBase.AbstractODEAlgorithm,
+    group_size::Integer;
+    continuity_term::Real=100,
+    kwargs...
+)
+    # Continuity loss between last state in previous prediction
+    # and current initial condition in ode_data
+    function continuity_loss(û_end, u_0)
+        return sum(abs, û_end - u_0)
+    end
```
Comment (ChrisRackauckas) [inline src/multiple_shooting.jl:243]: Define this as a generic function so it precompiles.

File: src/multiple_shooting.jl (hunk 2 above)
Comment (vboussange) [reply src/multiple_shooting.jl]: I have removed  the type specification `::Function`. Is that what you had in mind?

File: src/multiple_shooting.jl (hunk 3 above)
Comment (vboussange) [reply src/multiple_shooting.jl]: I am not sure what you mean by that.

Comment (ChrisRackauckas) [reply src/multiple_shooting.jl]: the reason for this dispatch is to handle the optional argument, but it really shouldn't be one of the ones in the middle then.

**Pull request merged**